### PR TITLE
MDEV-12914: CREATE TEMPORARY TABLE t LIKE t1 is substituted with MyIsam when using Rocksdb

### PR DESCRIPTION
**Pull request by an3l**

### CODE REVIEW

File: sql/sql_table.cc
Comment (vuvova) [inline sql/sql_table.cc:10679]: what? It seems this will **always** create a new table in MyISAM???

File: sql/sys_vars.cc
Comment (vuvova) [inline sql/sys_vars.cc:645]: the engine cannot be unknown, `Sys_var_plugin` will guarantee that the engine is installed and active.

I'd just use
```c
my_error(ER_ILLEGAL_HA_CREATE_OPTION, MYF(0), hton_name(hton)->str, "TEMPORARY");
```

File: sql/sys_vars.cc
Comment (vuvova) [inline sql/sys_vars.cc:646]: and no `sql_print_error` and `my_error(1)`, please, the above `my_error(ER_ILLEGAL_HA_CREATE_OPTION)` should be enough

File: sql/sys_vars.cc
Comment (vuvova) [inline sql/sys_vars.cc:636]: please, rename to something more specific. For example, `check_engine_supports_temporary`

**Review by vuvova — CHANGES_REQUESTED** [review]

File: mysql-test/suite/sys_vars/r/sysvars_star.result
Comment (cvicentiu) [inline mysql-test/suite/sys_vars/r/sysvars_star.result:63]: Rebase on latest 10.7. Serg has fixed this in a follow up commit.

In general, if this is something you need to fix, it needs to go in a separate commit, because it is not related to your push.

File: mysql-test/suite/sys_vars/r/default_tmp_storage_engine_rocksdb.result
Comment (cvicentiu) [inline mysql-test/suite/sys_vars/r/default_tmp_storage_engine_rocksdb.result:44]: Please change this to be:

`CREATE TEMPORARY TABLE t2 (a int);`

Do not rely on the implementation to check first that the engine can support temp tables, then if the table can be sanely created without columns.

File: mysql-test/suite/sys_vars/r/default_tmp_storage_engine_rocksdb.result
Comment (cvicentiu) [inline mysql-test/suite/sys_vars/r/default_tmp_storage_engine_rocksdb.result:46]: After this, you need to add the following line:
`CREATE TEMPORARY TABLE t2 (a int);`

Without this, you are not actually testing the behavior that `default_tmp_storage_engine=innodb;` actually tries to create the table with InnoDB.

You can also use Aria instead of InnoDB.

File: mysql-test/suite/sys_vars/t/default_tmp_storage_engine_rocksdb.test
Comment (cvicentiu) [inline mysql-test/suite/sys_vars/t/default_tmp_storage_engine_rocksdb.test:44]: Usually sourcing happens at the start of the test case, to make it more obvious that InnoDB is required. Why not use ARIA instead? For this purpose, it should be just fine and you don't need to bring innodb in.

File: sql/sql_table.cc
Comment (cvicentiu) [inline sql/sql_table.cc:11466]: Let's document this behavior change properly. Like we discussed in the KB.

File: sql/sys_vars.cc
Comment (cvicentiu) [inline sql/sys_vars.cc:748]: This is just a style nitpick. Let's move both line 748 and line 750 at the beginning of the function. After Line 745.

**Comment by cvicentiu** [comment]
Hi Anel! I think we're done with this one. I just left a few very simple comments related to coding style. Sorry I didn't articulate these better the first time. OK to push after you make those final tweaks, no reason to wait for an extra review from me.

File: mysql-test/suite/sys_vars/r/default_tmp_storage_engine_rocksdb.result
Comment (an3l) [reply mysql-test/suite/sys_vars/r/default_tmp_storage_engine_rocksdb.result:44]: Right, I was thinking is the right order satisfied, expecting to have syntax error first.

File: mysql-test/suite/sys_vars/r/default_tmp_storage_engine_rocksdb.result
Comment (an3l) [reply mysql-test/suite/sys_vars/r/default_tmp_storage_engine_rocksdb.result:46]: Ok

File: sql/sql_table.cc
Comment (an3l) [reply sql/sql_table.cc:11466]: Will do in https://mariadb.com/kb/en/create-table/#create-temporary-table

File: sql/sys_vars.cc
Comment (an3l) [reply sql/sys_vars.cc:748]: Thanks for the review @cvicentiu and for debugging sessions resulting in better quality and learning the new things.

File: sql/sys_vars.cc
Comment (vuvova) [inline sql/sys_vars.cc:752]: first argument should be `thd` here, not 0

File: sql/sys_vars.cc
Comment (vuvova) [inline sql/sys_vars.cc:763]: and here

File: sql/sys_vars.cc
Comment (vuvova) [inline sql/sys_vars.cc:760]: and here

**Review by vuvova — APPROVED** [review]

File: sql/sys_vars.cc
Comment (an3l) [reply sql/sys_vars.cc]: Ack, thanks @vuvova

**Comment by an3l** [comment]
Pushed with https://github.com/MariaDB/server/commit/f7216fa63d69448c3de1532a1dd197d0f28faefd to 10.7. Thanks

**Pull request closed without merging**
